### PR TITLE
Some corrections for custom timesteps

### PR DESCRIPTION
**Pull request by IlyaBeleckij**
Hi, I've found that your code is failing, then I needed to run script with custom number of ddim steps. So I found bugs and corrected them

### CODE REVIEW

File: ldm/models/diffusion/plms.py [hunk 1]
```diff
@@ -1017,7 +1028,7 @@ def plms_sampling_optimize_intrinsic_edit(
                     old_eps=old_eps,
                     t_next=ts_next,
                     input_image=input_image,
-                    optimizing_weight=torch.ones(50)[i],
+                    optimizing_weight=torch.ones(101)[i],
```
Comment (wuqiuche) [inline ldm/models/diffusion/plms.py:1031]: Hi IlyaBeleckij,

Thank you very much for your help! I see that the previous code is limited to a fixed denoising step of 50, and this pull request tries to support arbitrary denoising steps. I am still a little bit wondering about line 1301 of plms.py (here), which changes the fixed value of 50 to 101. Does this mean the code would support a fixed denoising step of 101 in this case?

**Comment by wuqiuche** [comment]
Hi IlyaBeleckij,

Thank you very much for your help! I see that the previous code is limited to a fixed denoising step of 50, and this pull request tries to support arbitrary denoising steps. I am still a little bit wondering about line 1301 of plms.py (I also commented in the code), which changes the fixed value of 50 to 101. Does this mean the code would support a fixed denoising step of 101 in this case?

File: ldm/models/diffusion/plms.py (hunk 1 above)
Comment (IlyaBeleckij) [reply ldm/models/diffusion/plms.py:1031]: Hi Wuqiuche, 
Thank you for your paper! It was awesome to get familiar with such concepts.
Yes, this is my fault from debugging. Honestly, it seems like we can just replace torch.ones(...)[i] with simple scalar one? Am I right?

Comment (wuqiuche) [reply ldm/models/diffusion/plms.py:1031]: Hi IlyaBeleckij,

Yeah you are right, my previous code was tedious here.. I will merge your pull request and just change it to scalar one here. Thank you very much for your help again!

**Pull request merged**